typescript/solver/solvers/onChain/filler.ts

```diff
@@ -0,0 +1,134 @@
+import { Wallet } from "@ethersproject/wallet";
+import { chainMetadata } from "@hyperlane-xyz/registry";
+import { MultiProvider } from "@hyperlane-xyz/sdk";
+import { bytes32ToAddress, ensure0x } from "@hyperlane-xyz/utils";
+
+import { MNEMONIC, PRIVATE_KEY } from "../../config.js";
+import { DestinationSettler__factory } from "../../contracts/typechain/factories/DestinationSettler__factory.js";
+import { Erc20__factory } from "../../contracts/typechain/factories/Erc20__factory.js";
+import { logDebug, logGreen } from "../../logger.js";
+import type { OpenEventArgs, ResolvedCrossChainOrder } from "../../types.js";
+import { getChainIdsWithEnoughTokens, settleOrder } from "./utils.js";
+
+export const create = () => {
```

Comment:
I'm curious how you thought about this functional approach vs. class-based approach. I'm generally a fan of the former, but it seems to me that the latter is more approachable. unless the framework you are using explicitly is built for that (like React)

Reply:
We just didn't want to get ahead of ourselves and start thinking about abstractions before we understood how things should work together so as not to box ourselves into structures that we would most likely have to change later. Definitely something we can look into though, but I'd tackle it once we have the general structure of the modules figured out.